### PR TITLE
Form block third-party features

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
#### Features
- Add reCaptcha from Google
- Email integration with Mailchimp and Sendinblue

### CODE REVIEW

**Comment by Soare-Robert-Daniel** [comment]
@irinelenache, @HardeepAsrani I need a test on this.

File: src/blocks/form/inspector.js [hunk 1]
```diff
@@ -91,21 +150,118 @@ const Inspector = ({
...
+				{
+					attributes.hasCaptcha && (
+						__( 'You can change the API Keys in Settings > Otter', 'otter-blocks' )
+					)
+				}
+
+	   		</PanelBody>
+			<PanelBody
+				title={ __( 'Integration', 'otter-blocks' )}
+				initialOpen={ true }
```
Comment (HardeepAsrani) [inline src/blocks/form/inspector.js:183]: initialOpen should be false.

File: src/frontend/form/index.js [hunk 2]
```diff
@@ -112,20 +183,58 @@ const collectAndSendInputFormData = ( form ) => {
...
+	inputContainer.classList.add( 'wp-block-themeisle-blocks-form-consent' );
+	container.insertBefore( inputContainer, button );
+
+	const input = document.createElement( 'input' );
+	input.type = 'checkbox';
+	input.name = 'consent';
+	input.id = 'consent';
+
+	const label = document.createElement( 'label' );
+	label.innerHTML = __( 'Share your email with third-party provider to recceive promotional offers!', 'otter-blocks' );
```
Comment (HardeepAsrani) [inline src/frontend/form/index.js:235]: This can be replaced with "I consent to my name and email being collected."

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested this PR but i see the integration doesn't work anymore. I get the email that contains the form details even if the **Action** is set to **Submit & Subscribe** or **Subscribe** and the new contacts aren't added. I tried both with Mailchimp and Sendinblue

**Comment by Soare-Robert-Daniel** [comment]
@irinelenache, @HardeepAsrani, now the integration should work. By refactoring the notification system I have introduced a lot of logical bugs

File: src/blocks/form/edit.js [hunk 3]
```diff
@@ -297,14 +300,16 @@ const Edit = ({
...
-						__( 'Integration detatils has been saved.', 'otter-blocks' ),
-						{
-							isDismissible: true,
-							type: 'snackbar'
-						}
-					);
+					if ( hasUpdatedNotice ) {
+						createNotice(
+							'info',
+							__( 'Integration detatils has been saved.', 'otter-blocks' ),
```
Comment (HardeepAsrani) [inline src/blocks/form/edit.js:306]: @Soare-Robert-Daniel Typo alert. :D

Comment (HardeepAsrani) [reply src/blocks/form/edit.js:306]: It should be:

Integration *details have* been saved.

**Comment by HardeepAsrani** [comment]
Newsletter works fine for me. If it's good for @irinelenache then we can merge.

File: inc/plugins/class-options-settings.php [hunk 4]
```diff
@@ -46,6 +46,30 @@ public function register_settings() {
 			)
 		);
 
+		register_setting(
+			'themeisle_blocks_settings',
+			'themeisle_google_captcha_api_site_key',
+			array(
+				'type'              => 'string',
+				'description'       => __( 'Google reCaptcha Site API key for the Google Maps Gutenberg Block.', 'otter-blocks' ),
```
Comment (HardeepAsrani) [inline inc/plugins/class-options-settings.php:54]: Form Block, not Google Maps.

File: inc/plugins/class-options-settings.php [hunk 5]
```diff
@@ -46,6 +46,30 @@ public function register_settings() {
...
+				'default'           => '',
+			)
+		);
+
+		register_setting(
+			'themeisle_blocks_settings',
+			'themeisle_google_captcha_api_secret_key',
+			array(
+				'type'              => 'string',
+				'description'       => __( 'Google reCaptcha Secret API key for the Google Maps Gutenberg Block.', 'otter-blocks' ),
```
Comment (HardeepAsrani) [inline inc/plugins/class-options-settings.php:66]: Form Block, not Google Maps.

File: src/blocks/form/edit.js [hunk 6]
```diff
@@ -75,7 +92,231 @@ const Edit = ({
...
+				saved = true;
+			}
+
+			setAPISaved( saved );
+			setGoogleCaptchaAPISecretKey( '' );
+			setGoogleCaptchaAPISiteKey( '' );
+
+			createNotice(
+				'info',
+				__( 'API Keys has been saved.', 'otter-blocks' ),
```
Comment (HardeepAsrani) [inline src/blocks/form/edit.js:244]: **have** not has.

File: src/blocks/form/save.js [hunk 7]
```diff
@@ -2,26 +2,37 @@
...
 			data-option-name={ attributes.optionName }
 		>
-			<div className="wp-block-themeisle-blocks-form__container">
+			<div className={'wp-block-themeisle-blocks-form__container'}>
 				<InnerBlocks.Content />
 
 				<div className="wp-block-button">
 					<button className="wp-block-button__link">
-						{ __( 'Submit', 'otter-blocks' ) }
+						{ __( 'subscribe' === attributes.action ? 'Subscribe' : 'Submit', 'otter-blocks' ) }
```
Comment (HardeepAsrani) [inline src/blocks/form/save.js:35]: They need to be wrapped in separate `__()` functions to work.

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested now and everything works fine 👍 🚀

**Pull request merged**